x/cdp/keeper/fees.go

```diff
@@ -45,11 +40,9 @@ func (k Keeper) UpdateFeesForRiskyCdps(ctx sdk.Context, collateralDenom string,
 	if priceDivLiqRatio.IsZero() {
 		priceDivLiqRatio = sdk.SmallestDec()
 	}
-	// NOTE - we have a fixed cutoff at 110% - this may or may not be changed in the future
-	normalizedRatio := sdk.OneDec().Quo(priceDivLiqRatio).Mul(sdk.MustNewDecFromStr("1.1"))
 
-	// now iterate over all the cdps based on collateral ratio
-	k.IterateCdpsByCollateralRatio(ctx, collateralDenom, normalizedRatio, func(cdp types.CDP) bool {
+	// now iterate over ALL the cdps
+	k.IterateAllCdps(ctx, func(cdp types.CDP) bool {
```

Comment:
This does not handle the case where the accumulated fees are rounded down to zero. Further, it needs to replicate the the functionality that was done previously by `HandleNewDebt`.

For each CDP:
1. Caluclate the additional fees (this is done)
2. Check if zero
3a. If zero, return false without updating cdp state
3b. If not zero, you need to mint debt coins to the cdp account and mint surplus coins divided between the liquidator and savings module accounts. Read the method from handle new debt and replicate it in the iteration loop here.  
3c. Before minting any coins, check if `newFeesSurplus` and `newFeesSavings` is rounded down to zero. If it is, return false without updating state.  
4. Update state at end of loop (currently, this is missing a call to keeper.SetCdp).

Reply:
🆗